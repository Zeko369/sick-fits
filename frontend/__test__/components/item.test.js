import { shallow } from "enzyme";

import ItemComponent from "../../components/Item";
import formatMoney from "../../lib/formatMoney";

const fakeItem = {
  id: "ASD123",
  title: "A Cool Item",
  price: 5000,
  description: "Foo",
  image: "dog.jpg",
  largeImage: "large-dog.jpg"
};

describe("<Item/>", () => {
  it("renders price and title properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const PriceTag = wrapper.find("PriceTag");
    const TitleTag = wrapper.find("Title a");

    expect(PriceTag.children().text()).toBe(formatMoney(fakeItem.price));
    expect(TitleTag.text()).toBe(fakeItem.title);
  });

  it("renders image properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const ImageTag = wrapper.find("img");

    expect(ImageTag.props().src).toBe(fakeItem.image);
    expect(ImageTag.props().alt).toBe(fakeItem.title);
  });

  it("renders buttons properly", () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const ButtonList = wrapper.find(".buttonList");

    expect(ButtonList.children()).toHaveLength(3);
    expect(ButtonList.find("Link").exists()).toBe(true);
    expect(ButtonList.find("AddToCart").exists()).toBe(true);
    expect(ButtonList.find("DeleteItem").exists()).toBe(true);
  });
});
