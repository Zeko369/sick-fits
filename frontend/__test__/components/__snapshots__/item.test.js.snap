// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matched the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ASD123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    Foo
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ASD123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ASD123"
      item={
        Object {
          "description": "Foo",
          "id": "ASD123",
          "image": "dog.jpg",
          "largeImage": "large-dog.jpg",
          "price": 5000,
          "title": "A Cool Item",
        }
      }
    />
    <DeleteItem
      id="ASD123"
    >
      Delete this item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
